File: TicTacToe/app/src/main/java/com/example/tictactoe/playersManager/Player.java
package com.example.tictactoe.playersManager;

public class Player {
    public int id;
    public String name;
    public int score;
    public String avatarPath;
    public Player(int id, String name, int score, String avatarPath){
        this.id = id;
        this.name = name;
        this.score = score;
        this.avatarPath = avatarPath;
    }
}

File: hw/1/TicTacToe/app/src/main/java/ru/brz/tictactoe/model/Player.kt
package ru.brz.tictactoe.model

import android.graphics.drawable.Drawable

data class Player (
    var name: String,
    var iconId: Drawable,
    var signId: Int,
    var score: Int
){
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (other !is Player) return false

        if (name != other.name) return false
        if (signId != other.signId) return false

        return true
    }

    override fun hashCode(): Int {
        var result = name.hashCode()
        result = 31 * result + signId
        return result
    }
}